src/parsing.jl

```diff
@@ -121,6 +127,12 @@ You can implement default PostgreSQL-specific parsing for a given type by overri
 """
 Base.parse(::Type{T}, pqv::PQValue) where {T} = pqparse(T, string_view(pqv))
 
+Base.parse(::Type{T}, pqv::PQValue{R, BINARY}) where {T <: Integer, R} = T(pqparse(
+    T,
+    Ptr{get(pqv.jl_result.type_lookup, R, String)}(data_pointer(pqv)),
+))
+
```

Comment:
```suggestion
for int_sym in (:int2, :int4, :int8)
    @eval function Base.parse(::Type{T}, pqv::PQValue{$(oid(int_sym)), BINARY}) where {T <: Number}
        return convert(T, ntoh(unsafe_load(Ptr{$(_DEFAULT_TYPE_MAP[int_sym])}(data_pointer(pqv))))
    end
end
```

This should move down to where the `pqparse(::Type{<:Integer}, pqv::Ptr)` method was added and replace that.

I'm not completely sure about what the boundary between `pqparse` and `parse` should be in the case of binary data so let's remove `pqparse` from this binary code path now and see whether it becomes a useful abstraction point later on. Perhaps we will end up with a method like this:
```julia
Base.parse(::Type{T}, pqv::PQValue{OID, BINARY}) where {T, OID} = pqparse(T, bytes_view(pqv))
```
so that people overriding `pqparse` get an iterable view of bytes. But maybe we just won't need it.